Mage.Sets/src/mage/cards/s/SyrixCarrierOfTheFlame.java

```diff
@@ -59,23 +64,23 @@ public SyrixCarrierOfTheFlame(UUID ownerId, CardSetInfo setInfo) {
         // target Phoenix you control deals damage equal to its power to any target.
         BeginningOfEndStepTriggeredAbility ability = new BeginningOfEndStepTriggeredAbility(
                 new DamageWithPowerFromOneToAnotherTargetEffect(),
-                TargetController.EACH_PLAYER,
+                TargetController.ANY,
                 SyrixCarrierOfTheFlameCondition.instance,
                 false
         );
         ability.addTarget(new TargetPermanent(phoenixFilter));
         ability.addTarget(new TargetAnyTarget());
-        ability.addWatcher(new SyrixCarrierOfTheFlameWatcher());
+        ability.addWatcher(new CardsLeftGraveyardWatcher());
         this.addAbility(ability);
 
         // Whenever another Phoenix you control dies, you may cast Syrix, Carrier of the Flame from your graveyard.
-        this.addAbility(new DiesCreatureTriggeredAbility(
-                Zone.GRAVEYARD,
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(new DiesCreatureTriggeredAbility(
```

Comment:
this is not a conditional intervening if, so this probably isn't the way to do it.

If [[Sproutback Trudge]] works correctly, move its effect to common `MayCastSourceFromGraveyardOneShotEffect` and reuse logic